pkg/clustermesh/remote_cluster.go

```diff
@@ -274,6 +276,53 @@ func (rc *remoteCluster) restartRemoteConnection(allocator RemoteIdentityWatcher
 	)
 }
 
+func (rc *remoteCluster) getClusterConfig(ctx context.Context, backend kvstore.BackendOperations, forceRequired bool) (*cmtypes.CiliumClusterConfig, error) {
+	var (
+		err                       error
+		config                    *cmtypes.CiliumClusterConfig
+		requireConfig             = forceRequired
+		clusterConfigPollInterval = 30 * time.Second
+		clusterConfigPollTimeout  = 3 * time.Minute
+	)
+
+	if !requireConfig {
+		// Let's check whether the kvstore states that the cluster configuration should be always present.
+		requireConfig, err = IsClusterConfigRequired(ctx, backend)
+		if err != nil {
+			return nil, fmt.Errorf("failed to detect whether the cluster configuration is required: %w", err)
+		}
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, clusterConfigPollTimeout)
+	defer cancel()
+
+	// We retry here rather than returning an simply returning an error and relying on the
+	// controller backoff period to avoid recreating every time a new connection to the remote
+	// kvstore, which would introduce an unnecessary overhead. Still, we do return in case of
+	// consecutive failures, to ensure that we do not retry forever is something strange happened.
+	err = wait.PollImmediateInfiniteWithContext(ctx, clusterConfigPollInterval, func(ctx context.Context) (done bool, err error) {
```

Comment:
Any reason why we don't use [our controllers](https://github.com/cilium/cilium/blob/main/pkg/controller/manager.go) for this? They have the benefit of showing up in `cilium status` if something fails. With ` wait.PollImmediateInfiniteWithContext` the failures will get hidden AFAICT.

Reply:
Essentially, the remote cluster logic is already wrapped by a controller. The idea here was to have a retry mechanism which doesn't cause that controller to restart (by simply returning the error there), because it would also cause the restart of the etcd connection, introducing an unnecessary overhead. But yeah, it makes sense to use another controller also for that rather than `wait.PollImmediateInfiniteWithContext`.

Reply:
I've updated the PR to use the controller. @aanm PTAL